client/snap_op.go

```diff
@@ -39,6 +39,7 @@ type SnapOptions struct {
 	Dangerous        bool   `json:"dangerous,omitempty"`
 	IgnoreValidation bool   `json:"ignore-validation,omitempty"`
 	Unaliased        bool   `json:"unaliased,omitempty"`
+	Instance         string `json:"instance,omitempty"`
```

Comment:
For installations, specifically, we may want to provide a name even when we're not actually providing the instance key, because of the delta between what's reported inside the snap as a name and what's reported in the snap-declaration assertion.

Even besides that, there's a question about how we plan to handle the "name" in the API. We have existing APIs that already offer a "name" field. Are we going to report instance names on those, to remain compatible, or will we keep the snap name alone there, and have a separate field. We should make a consistent rule and use the same guideline here.

Reply:
I think we should use `name`. This will be either `<snap>` or `<snap>_<instance-key>`, and should 
be entirely transparent for the client side. That is, I plan to drop this particular `Instance` field and change `client.InstallPath()` to have the name passed explicitly there. I'll be pushing the change soon.

Using name in the client API is a bit confusing at times. For instance, when installing a single snap the client will do `POST /v2/snaps/<snap-name>`, the name is already passed explicitly in the URL. When installing multiple snaps, it does `POST /v2/snaps`, with snap names passed in body as `snaps` field. Then when installing from path, the client does a request to `POST /v2/snaps`, with `multipart/form-data`, and will pass `name` form field with the snap name. In all cases except installation from snap file the name is passed in some form.